impl/src/planning/subplanners.rs

```diff
@@ -122,6 +122,8 @@ impl<'planner> WorkspaceSubplanner<'planner> {
       GenMode::Vendored => {
         checks::check_all_vendored(self.crate_catalog.entries(), &self.settings.workspace_path)?;
       },
+      // Settings should always have `genmode` set to one of the above fields
+      GenMode::Unspecified => unreachable!(),
```

Comment:
We can just do nothing here, rather than "unreachable!()".

Reply:
I feel like this communicates the intent though can you share more on why you're feeling it would better to do nothing over `unreachable`?

Reply:
This might have been answered in your other comment

Reply:
For me intent is secondary to defensive programming. I interpret defensive programming in this case to mean "do the least surprising thing", and "don't expect upstream code to stay static (meaning unchanging)".

That tells me two things. Since the validation logic is (somewhat far) upstream of us, to be defensive, we should handle all possible inputs within reason. This variant is an explicit possible variant, and there's really very little preventing future changes to the code from toggling genmode into this state for reasons we can't currently predict.

This particular case is my favorite kind of case, because we can do "nothing" and the behavior won't be surprising. In this case, the switch is only used to enable special per-genmode validation logic. Since we're not generating anything, we have no validation to do.